### PR TITLE
Add checks for api dates

### PR DESCRIPTION
**Pull request by TinasheMzondiwa**
# What did you change:

Added checks for invalid dates coming from lesson data like `"31/04/2021"`

# Screenshot/GIFs of this change

_Quick glance of what is changing where_

# Merge checklist

- [ ] Automated (unit/ui) tests
- [x] Manual tests

### CODE REVIEW

File: features/lessons/src/main/java/com/cryart/sabbathschool/lessons/ui/readings/SSReadingViewModel.kt [hunk 1]
```diff
@@ -169,6 +168,16 @@ class SSReadingViewModel(
             })
     }
 
+    private fun parseDate(date: String): DateTime? {
+        return try {
+            DateTimeFormat.forPattern(SSConstants.SS_DATE_FORMAT)
+                .parseLocalDate(date).toDateTimeAtStartOfDay()
+        } catch (ex: Exception) {
+            Timber.e(ex)
```
Comment (merlin-zaraza) [inline features/lessons/src/main/java/com/cryart/sabbathschool/lessons/ui/readings/SSReadingViewModel.kt:176]: Please include `date` in Timber message so we can find it in content.

File: features/lessons/src/main/java/com/cryart/sabbathschool/lessons/ui/readings/SSReadingViewModel.kt [hunk 2]
```diff
@@ -285,11 +294,16 @@ class SSReadingViewModel(
...
-                    .parseLocalDate(date)
-            ).capitalize(Locale.getDefault())
+        return try {
+            DateTimeFormat.forPattern(DateFormatOutput)
+                .print(
+                    DateTimeFormat.forPattern(SSConstants.SS_DATE_FORMAT)
+                        .parseLocalDate(date)
+                ).capitalize(Locale.getDefault())
+        } catch (ex: IllegalArgumentException) {
+            Timber.e(ex)
```
Comment (merlin-zaraza) [inline features/lessons/src/main/java/com/cryart/sabbathschool/lessons/ui/readings/SSReadingViewModel.kt:304]: Please include `date` in Timber message so we can find it in content.

**Review by merlin-zaraza — APPROVED** [review]
Good, just few comments!

File: features/lessons/src/main/java/com/cryart/sabbathschool/lessons/ui/readings/SSReadingViewModel.kt (hunk 1 above)
Comment (TinasheMzondiwa) [reply features/lessons/src/main/java/com/cryart/sabbathschool/lessons/ui/readings/SSReadingViewModel.kt:176]: The date will be in the exception message.
It looks like this:
`Cannot parse "31/04/2021": Value 31 for dayOfMonth must be in the range [1,30]`

File: features/lessons/src/main/java/com/cryart/sabbathschool/lessons/ui/readings/SSReadingViewModel.kt (hunk 2 above)
Comment (TinasheMzondiwa) [reply features/lessons/src/main/java/com/cryart/sabbathschool/lessons/ui/readings/SSReadingViewModel.kt:304]: Same as above.

**Pull request merged**